src/demo/app.ts

```diff
@@ -92,21 +95,34 @@ fin.desktop.main(async () => {
         });
     });
 
-    addEventListener('notification-created', (event: NotificationCreatedEvent) => {
-        logMessage(`CREATE action received from notification ${event.notification.id}`);
-    });
-    addEventListener('notification-closed', (event: NotificationClosedEvent) => {
-        logMessage(`CLOSE action received from notification ${event.notification.id}`);
-    });
-    addEventListener('notification-action', (event: NotificationActionEvent) => {
-        const {notification, trigger, control} = event;
+    const queryParams = new URLSearchParams(location.search);
+    if (queryParams.get('inttest') === null) {
+        addEventListener('notification-created', (event: NotificationCreatedEvent) => {
+            logMessage(`CREATE action received from notification ${event.notification.id}`);
+        });
+        addEventListener('notification-closed', (event: NotificationClosedEvent) => {
+            logMessage(`CLOSE action received from notification ${event.notification.id}`);
+        });
+        addEventListener('notification-action', (event: NotificationActionEvent) => {
+            const {notification, trigger, control} = event;
 
-        if (trigger === 'select') {
-            logMessage(`SELECT action received from notification ${event.notification.id}`);
-        } else if (control && control.type === 'button') {
-            const buttonIndex = notification.buttons.indexOf(control);
+            if (trigger === 'select') {
+                logMessage(`SELECT action received from notification ${event.notification.id}`);
+            } else if (control && control.type === 'button') {
+                const buttonIndex = notification.buttons.indexOf(control);
 
-            logMessage(`CONTROL action on button ${control.title} (Index: ${buttonIndex}) on notification ${notification.id}`);
-        }
-    });
+                logMessage(`CONTROL action on button ${control.title} (Index: ${buttonIndex}) on notification ${notification.id}`);
+            }
+        });
+    } else if (queryParams.get('inttest') === 'listeners-on-startup') {
```

Comment:
Love this change. Much better way of handling it.

Reply:
✅